sdk/storage/azure_storage_blob/src/clients/blob_client.rs

```diff
@@ -243,4 +244,20 @@ impl BlobClient {
         let response = block_blob_client.get_block_list(list_type, options).await?;
         Ok(response)
     }
+
+    /// Sets the tier on a blob. Standard tiers are only applicable for Block blobs, while Premium tiers are only applicable
+    /// for Page blobs.
+    ///
+    /// # Arguments
+    ///
+    /// * `tier` - The tier to be set on the blob.
+    /// * `options` - Optional configuration for the request.
+    pub async fn set_tier(
+        &self,
+        tier: AccessTier,
+        options: Option<BlobClientSetTierOptions<'_>>,
+    ) -> Result<Response<()>> {
+        let response = self.client.set_tier(tier, options).await?;
+        Ok(response)
```

Comment:
Should just be able to call: `self.client.set_tier(tier, options).await`

Reply:
Good callout here- will apply this in a sweeping change next PR to all applicable APIs